### PR TITLE
Add proper parser for url() function

### PR DESCRIPTION
**Pull request by dirkschulze**
## Description

Add proper parser for url() function and improve results. Also add new test.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dirkschulze** [comment]
@moazin Could you take a look at this PR please?

File: svgnative/test/paint.svg [hunk 1]
```diff
@@ -42,12 +42,13 @@
     </g>
     <rect x="140" y="40" width="20" height="20" fill="#F00 red" stroke="#F00 red"/>
     <rect x="160" y="40" width="20" height="20" fill="#F00 #F00" stroke="#F00 #F00"/>
-    <rect x="180" y="40" width="20" height="20" fill="url(#notPresent) #008000" stroke="url(#notPresent) #008000"/>
-    <rect x="0" y="60" width="20" height="20" fill="url(#notPresent) green" stroke="url(#notPresent) green"/>
+    <rect x="180" y="40" width="20" height="20" fill="url(#notPresent) #008000" stroke=" url(#notPresent) #008000"/>
```
Comment (vnair81) [inline svgnative/test/paint.svg:45]: Is this spacing change relevant?

File: svgnative/test/paint.svg [hunk 2]
```diff
@@ -42,12 +42,13 @@
...
+    <rect x="0" y="60" width="20" height="20" fill=" url(#notPresent) green" stroke="url(#notPresent) green"/>
```
Comment (vnair81) [inline svgnative/test/paint.svg:46]: Is this spacing change relevant?

File: svgnative/test/paint.txt [hunk 3]
```diff
@@ -100,6 +100,9 @@
             [path Rect(80,60,20,20)
                 fill: {hasFill: true winding: nonzero paint: rgba(0,0,0,1)}
                 stroke: {hasStroke: false width: 1 cap: butt join: miter miter: 4 dashOffset: 0 paint: rgba(0,0,0,1)}]
+            [path Rect(80,60,20,20)
```
Comment (vnair81) [inline svgnative/test/paint.txt:103]: This just seems like we duplicated above three lines.
Intentional?

File: svgnative/test/paint.svg (hunk 1 above)
Comment (dirkschulze) [reply svgnative/test/paint.svg:45]: Yes, the issue was discovered recently by adding whitespaces at the beginning. Changed the 2 rectangle examples to cover a whitespace at the start of an url() on fill and stroke.

File: svgnative/test/paint.svg (hunk 2 above)
Comment (dirkschulze) [reply svgnative/test/paint.svg:46]: Yes, it is testing the improvement of the refactoring+bug fixing.

**Review by vnair81 — APPROVED** [review]

**Pull request merged**